From 8aff4298d3f64d0163ede2c0618292de449d818c Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Sat, 14 Dec 2024 02:29:45 +0900
Subject: [PATCH 1/2] Fix WASI build of `_copyDirectoryMetadata` (#1094)

Extended attributes don't exist in WASI, so we need to exclude the use
of xattr-related APIs including `flistxattr`.
---
 Sources/FoundationEssentials/FileManager/FileOperations.swift | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Sources/FoundationEssentials/FileManager/FileOperations.swift b/Sources/FoundationEssentials/FileManager/FileOperations.swift
index 83d131a..ce4de44 100644
--- a/Sources/FoundationEssentials/FileManager/FileOperations.swift
+++ b/Sources/FoundationEssentials/FileManager/FileOperations.swift
@@ -911,6 +911,7 @@ enum _FileOperations {
     
     #if !canImport(Darwin)
     private static func _copyDirectoryMetadata(srcFD: CInt, srcPath: @autoclosure () -> String, dstFD: CInt, dstPath: @autoclosure () -> String, delegate: some LinkOrCopyDelegate) throws {
+        #if !os(WASI)
         // Copy extended attributes
         var size = flistxattr(srcFD, nil, 0)
         if size > 0 {
@@ -936,6 +937,7 @@ enum _FileOperations {
                 }
             }
         }
+        #endif
         var statInfo = stat()
         if fstat(srcFD, &statInfo) == 0 {
             // Copy owner/group
-- 
2.46.0

